nnvm/src/top/nn/convolution.cc

```diff
@@ -79,7 +79,15 @@ inline bool Conv2DInferShape(const nnvm::NodeAttrs& attrs,
                  param.kernel_size[1]});
 
   wshape = ConvertLayout(wshape, kOIHW, kernel_layout);
-  wshape[0] *= param.groups;
+
+  // Depthwise
+  // NCHW : Expects weights in CNHW - Conversion is handled in frontend.
+  // NHWC : Original format (HWCN)
+  if (param.layout == "NHWC") {
+    wshape[kernel_layout.indexof('I')] *= param.groups;
+  } else {
+    wshape[0] *= param.groups;
```

Comment:
when param.layout == "NCHW", kernel_layout.indexof('I') = 0?

Reply:
because the weights format for NCHW sent as CHNW from front end and expectation is also same at compute.
Ref.

Frontend: 
https://github.com/dmlc/tvm/blob/b877687fbeb8a6c12b62aac9869f7e54091395f4/nnvm/python/nnvm/frontend/keras.py#L145

Compute:
https://github.com/dmlc/tvm/blob/b877687fbeb8a6c12b62aac9869f7e54091395f4/topi/python/topi/nn/depthwise_conv2d.py#L38

Reply:
I think ``wshape[kernel_layout.indexof('O')] *= param.groups`` can handle both cases
You can try some tests

Reply:
No it doesn't work this way as for kernel layout for NCHW deviated from the format. (it's actually CHNW passed from frontend and same is expected in compute.).